x-pack/plugins/infra/server/services/inventory_views/inventory_views_client.ts

```diff
@@ -0,0 +1,199 @@
+
+    // Validate there is not a view with the same name
+    await this.assertNameConflict(attributes.name);
+
+    const inventoryViewSavedObject = await this.savedObjectsClient.create(
+      inventoryViewSavedObjectName,
+      attributes
+    );
+
+    return this.mapSavedObjectToInventoryView(inventoryViewSavedObject);
+  }
+
+  public async update(
+    inventoryViewId: string,
+    attributes: CreateInventoryViewAttributesRequestPayload,
+    query: InventoryViewRequestQuery
+  ): Promise<InventoryView> {
+    this.logger.debug(`Trying to update inventory view with id "${inventoryViewId}"...`);
+
+    // Validate there is not a view with the same name
+    await this.assertNameConflict(attributes.name, [inventoryViewId]);
+
+    const sourceId = query.sourceId ?? 'default';
+
+    const [sourceConfiguration, inventoryViewSavedObject] = await Promise.all([
+      this.infraSources.getSourceConfiguration(this.savedObjectsClient, sourceId),
+      this.savedObjectsClient.update(inventoryViewSavedObjectName, inventoryViewId, attributes),
+    ]);
+
+    return this.mapSavedObjectToInventoryView(
+      inventoryViewSavedObject,
+      sourceConfiguration.configuration.inventoryDefaultView
+    );
+  }
+
+  public delete(inventoryViewId: string): Promise<{}> {
+    this.logger.debug(`Trying to delete inventory view with id ${inventoryViewId} ...`);
+
+    return this.savedObjectsClient.delete(inventoryViewSavedObjectName, inventoryViewId);
+  }
+
+  private mapSavedObjectToInventoryView(
+    savedObject: SavedObject | SavedObjectsUpdateResponse,
+    defaultViewId?: string
+  ) {
+    const inventoryViewSavedObject = decodeOrThrow(inventoryViewSavedObjectRT)(savedObject);
+
+    return {
+      id: inventoryViewSavedObject.id,
+      version: inventoryViewSavedObject.version,
+      updatedAt: inventoryViewSavedObject.updated_at,
+      attributes: {
+        ...inventoryViewSavedObject.attributes,
+        isDefault: inventoryViewSavedObject.id === defaultViewId,
+        isStatic: false,
+      },
+    };
+  }
+
+  private moveDefaultViewOnTop(views: InventoryView[]) {
+    const defaultViewPosition = views.findIndex((view) => view.attributes.isDefault);
+
+    if (defaultViewPosition !== -1) {
+      const element = views.splice(defaultViewPosition, 1)[0];
+      views.unshift(element);
+    }
+
+    return views;
+  }
+
+  /**
+   * We want to control conflicting names on the views
+   */
+  private async assertNameConflict(name: string, whitelist: string[] = []) {
+    const results = await this.savedObjectsClient.find<InventoryViewAttributes>({
+      type: inventoryViewSavedObjectName,
+      perPage: 1000,
+    });
+
+    const hasConflict = [InventoryViewsClient.createStaticView(), ...results.saved_objects].some(
+      (obj) => !whitelist.includes(obj.id) && obj.attributes.name === name
+    );
+
+    if (hasConflict) {
+      throw Boom.conflict('A view with that name already exists.');
+    }
+  }
```

Comment:
**note:** we can't delegate this search to the `savedObjectsClient.find` method since, being the attributes mapping set to `dynamic: false`, searching for the name won't return results.